frontend/ui/src/app/licenses/edit-license.component.ts

```diff
@@ -0,0 +1,267 @@
+      const val = this.editForm.getRawValue();
+      if (!val.includeAllVersions) {
+        this.versionsSelected = val.versions.filter((v) => !!v).length;
+      }
+      if (this.editForm.valid) {
+        this.onChange({
+          id: val.id,
+          name: val.name,
+          expiresAt: val.expiresAt ? new Date(val.expiresAt) : undefined,
+          applicationId: val.applicationId,
+          versions: this.getSelectedVersions(val.includeAllVersions!, val.versions ?? []),
+          ownerUserAccountId: val.ownerUserAccountId,
+          registryUrl: val.registry.url?.trim() || undefined,
+          registryUsername: val.registry.username?.trim() || undefined,
+          registryPassword: val.registry.password?.trim() || undefined,
+        });
+      } else {
+        this.onChange(undefined);
+      }
+    });
+    this.editForm.controls.applicationId.valueChanges
+      .pipe(
+        takeUntil(this.destroyed$),
+        switchMap(async (applicationId) => {
+          const apps = await firstValueFrom(this.applicationsService.list());
+          return apps.find((a) => a.id === applicationId);
+        })
+      )
+      .subscribe((selectedApplication) => {
+        this.versionsArray.clear({emitEvent: false});
+        const applicationVersions = selectedApplication?.versions ?? [];
+        const licenseVersions = this.license()?.versions;
+        let anySelected = false;
+        for (let i = 0; i < applicationVersions.length; i++) {
+          const version = applicationVersions[i];
+          const selected = !!licenseVersions?.some((v) => v.id === version.id);
+          this.versionsArray.push(this.fb.control(selected), {emitEvent: i === applicationVersions.length - 1});
+          anySelected = anySelected || selected;
+        }
+        if (!anySelected) {
+          this.editForm.controls.includeAllVersions.patchValue(true);
+        }
+        this.selectedApplication.set(selectedApplication);
+      });
+  }
+
+  private getSelectedVersions(includeAllVersions: boolean, versionControls: (boolean | null)[]): ApplicationVersion[] {
+    if (includeAllVersions) {
+      return [];
+    }
+    const app = this.selectedApplication();
+    return versionControls
+      .map((v, idx) => {
+        if (v) {
+          return app?.versions?.[idx];
+        }
+        return undefined;
+      })
+      .filter((v) => !!v);
+  }
+
+  ngAfterViewInit() {
+    // from https://github.com/angular/angular/issues/45089
+    this.injector
+      .get(NgControl)
+      .control!.events.pipe(takeUntil(this.destroyed$))
+      .subscribe((event) => {
+        if (event instanceof TouchedChangeEvent) {
+          if (event.touched) {
+            this.editForm.markAllAsTouched();
+          }
+        }
+      });
+  }
+
+  toggleDropdown() {
+    this.dropdownOpen.update((v) => !v);
+    if (this.dropdownOpen()) {
+      this.dropdownWidth = (this.dropdownTriggerButton.nativeElement as Element).getBoundingClientRect().width;
```

Comment:
combined with previous suggestion
```suggestion
      this.dropdownWidth = this.dropdownTriggerButton.nativeElement.getBoundingClientRect().width;
```